app/models/account/buyer_methods.rb

```diff
@@ -161,4 +161,8 @@ def paying_monthly!
   def not_paying_monthly!
     settings.update_attribute(:monthly_charging_enabled, false)
   end
+
+  def account_suspended_or_scheduled_for_deletion?(account)
```

Comment:
@guicassolato is this what you were thinking?

Reply:
Almost. I meant https://github.com/3scale/porta/pull/2130/commits/cfb4720fd2f6e3ff8d866d9776d6e9277f5ce540.

1. Since the module (`Account::BuyerMethods`, but also `Account::States`) is included in `Account`, the account is the context. No need to pass it as argument 😉
2. `suspended_or_scheduled_for_deletion?` has strictly to do with the state of the account, so I think better being part of `Account::States` than `Account::BuyerMethods`)